File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.cpp
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {

        for(int i=0;i<nums2.size();i++){
            nums1.push_back(nums2[i]);
        }
        sort(nums1.begin(),nums1.end());

        if(nums1.size()%2==0){ // it means this sorted array has two element in the middle 
            int second_mid = nums1.size()/2;
            int first_mid = second_mid-1;

            int f_m = nums1[first_mid];
            int s_m = nums1[second_mid];

            double val = (f_m + s_m)/2.0;

            return val;
            
        }else{// means array length is odd
            return nums1[nums1.size()/2];
        }

        return 2.9;
    }
};

File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int[] ans = mergedSortedArray(nums1,nums2);

        if(ans.length % 2 == 0) {
            double ans2 = (double)(ans [ans.length/2]  + ans [ans.length/2 -1])/2;
            return ans2;
        } else {
            double ans2 = (double) ( ans [ans.length/2]);
            return ans2;
        }

    }
    static int[] mergedSortedArray(int[] arr1, int[] arr2) {
        int n1 = arr1.length;
        int n2 = arr2.length;

        int[] mergedArray = new int[n1 + n2];
        int i = 0, j = 0, k = 0;

        while(i < n1 && j < n2) {
            if(arr1[i] <= arr2[j]) {
                mergedArray[k++] = arr1[i++];                
            } else {
                mergedArray[k++] = arr2[j++];
            }
        }

        while( i < n1) {
            mergedArray[k++] = arr1[i++];
        }

        while( j < n2) {
            mergedArray[k++] = arr2[j++];
        }


        return mergedArray;
    }
}